packages/utilities/src/lib/objectEntries.ts

```diff
@@ -0,0 +1,5 @@
+import type { NonNullObject } from './utilityTypes';
+
+export function objectEntries<T extends NonNullObject>(obj: T): T extends ArrayLike<any> ? [`${number}`, T[number]][] : [keyof T, T[keyof T]][] {
+	return Object.entries(obj) as T extends ArrayLike<any> ? [`${number}`, T[number]][] : [keyof T, T[keyof T]][];
```

Comment:
comment on #474 is a good one

```suggestion
	return Object.entries(obj) as T extends ArrayLike<infer Values> ? Values[] : [keyof T, T[keyof T]][];
```

<img width="684" alt="image" src="https://user-images.githubusercontent.com/4019718/193416894-da1a9610-98f9-4c7a-b2c6-227989130baa.png">
<img width="722" alt="image" src="https://user-images.githubusercontent.com/4019718/193416901-06fe9e82-3310-4424-9dda-d04c01b20c20.png">